server/src/main/java/org/elasticsearch/search/lookup/SearchLookup.java

```diff
@@ -53,14 +53,22 @@ public class SearchLookup {
      */
     public SearchLookup(
         Function<String, MappedFieldType> fieldTypeLookup,
-        TriFunction<MappedFieldType, Supplier<SearchLookup>, MappedFieldType.FielddataOperation, IndexFieldData<?>> fieldDataLookup
+        TriFunction<MappedFieldType, Supplier<SearchLookup>, MappedFieldType.FielddataOperation, IndexFieldData<?>> fieldDataLookup,
+        SourceLookup sourceLookup
```

Comment:
I think I'd prefer to pass the `SourceProvider` as a parameter here, rather than the `SourceLookup` directly